packages/host/app/resources/search.ts

```diff
@@ -66,6 +68,17 @@ export class SearchResource extends Resource<Args> {
     this.loaded = this.search.perform(query);
 
     if (isLive) {
+      // need to unsubscribe the old query before subscribing the new query
+      if (
+        this.subscriptions &&
+        (this.#previousQuery !== query || this.#previousRealms !== realms)
+      ) {
+        for (let subscription of this.subscriptions) {
+          subscription.unsubscribe();
+        }
+      }
+      this.#previousQuery = query;
+      this.#previousRealms = realms;
```

Comment:
not unsubscribing live searches when the query changes compounded the issue and resulted in geometric expansion of the queries being issued in this scenario